server/src/main/java/org/opensearch/telemetry/tracing/listener/TraceableSearchRequestOperationsListener.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.telemetry.tracing.listener;
+
+import org.opensearch.action.search.SearchPhaseContext;
+import org.opensearch.action.search.SearchRequestContext;
+import org.opensearch.action.search.SearchRequestOperationsListener;
+import org.opensearch.telemetry.tracing.AttributeNames;
+import org.opensearch.telemetry.tracing.ScopedSpan;
+import org.opensearch.telemetry.tracing.Span;
+import org.opensearch.telemetry.tracing.SpanBuilder;
+import org.opensearch.telemetry.tracing.SpanContext;
+import org.opensearch.telemetry.tracing.Tracer;
+
+import static org.opensearch.core.common.Strings.capitalize;
+
+/**
+ * SearchRequestOperationsListener subscriber for search request tracing
+ *
+ * @opensearch.internal
+ */
+public final class TraceableSearchRequestOperationsListener extends SearchRequestOperationsListener {
+    private final Tracer tracer;
+    private final Span requestSpan;
+    private ScopedSpan phaseScopedSpan;
+
+    public TraceableSearchRequestOperationsListener(final Tracer tracer, final Span requestSpan) {
+        this.tracer = tracer;
+        this.requestSpan = requestSpan;
+        this.phaseScopedSpan = null;
+    }
+
+    @Override
+    protected void onPhaseStart(SearchPhaseContext context) {
+        assert phaseScopedSpan == null : "There should be only one search phase active at a time";
+        phaseScopedSpan = tracer.startScopedSpan(
+            SpanBuilder.from(capitalize(context.getCurrentPhase().getName()), new SpanContext(requestSpan))
+        );
+    }
+
+    @Override
+    protected void onPhaseEnd(SearchPhaseContext context, SearchRequestContext searchRequestContext) {
+        assert phaseScopedSpan != null : "There should be a search phase active at that time";
+        phaseScopedSpan.close();
+        phaseScopedSpan = null;
+    }
+
+    @Override
+    protected void onPhaseFailure(SearchPhaseContext context, Throwable cause) {
+        phaseScopedSpan.setError(new Exception(cause));
```

Comment:
Apologies, I thought you would do the same along with `onPhaseEnd` but better be safe:
```suggestion
assert phaseScopedSpan != null : "There should be a search phase active at that time";
```